### DISCUSSION TITLE
Create .peg/clean.peg files from parsers written in python.

### DISCUSSION THREAD

**Discussion by MaxwellWibert** [textX/Arpeggio#119] [Ideas]
It would be pretty useful to be able to programmatically take a parser defined using Python syntax and write the grammar to a .peg or clean.peg file. Better yet if you could write into other PEG formats by specifying the grammar of those formats with an additional peg file (assuming it interfaces similarly to the arpeggio.peg file).

**Comment by igordejanovic** [comment]
It should be straightforward to do as the Arpeggio takes the grammar and represent it in memory as a "parser model" which corresponds 1:1 to the grammar provided. You just need to visit the parser model nodes and generate the string with the correct syntax.

But, I think this feature is outside of the scope of the project. Let me know if you do it and publish as open source so I can link to it from the docs.
